The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11706539046/job/32603891798) [(plain)](https://github.com/rust-lang/rust/commit/62c0c994492c7f79a745619b06f1f7849c5f2ad9/checks/32603891798/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.729 Building wheels for collected packages: reuse
#13 2.730   Building wheel for reuse (pyproject.toml): started
#13 2.980   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 2.981   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 2.981   Stored in directory: /tmp/pip-ephem-wheel-cache-xra7493d/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 2.984 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.382 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.382 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.919 Collecting virtualenv
#13 3.919 Collecting virtualenv
#13 3.963   Downloading virtualenv-20.27.1-py3-none-any.whl (3.1 MB)
#13 4.025      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 52.5 MB/s eta 0:00:00
#13 4.078 Collecting platformdirs<5,>=3.9.1
#13 4.082   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.117 Collecting filelock<4,>=3.12.2
#13 4.140 Collecting distlib<1,>=0.3.7
#13 4.145   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.153      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 82.1 MB/s eta 0:00:00
#13 4.153      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 82.1 MB/s eta 0:00:00
#13 4.232 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.407 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.27.1
#13 DONE 4.5s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      300992 kB
DirectMap2M:     7038976 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://static.rust-lang.org/dist/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 31.71s
##[endgroup]
fmt check
Diff in /checkout/library/std/src/f128/tests.rs:2:
 #![cfg(reliable_f128)]
 use crate::f128::consts;
+use crate::num::FpCategory as Fp;
+use crate::num::FpCategory as Fp;
 #[cfg(not(reliable_f128))]
 use crate::num::*;
-use crate::num::FpCategory as Fp;
 // Note these tolerances make sense around zero, but not for more extreme exponents.
 
 
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/sync/mpmc/select.rs" "/checkout/library/std/src/sync/mpmc/zero.rs" "/checkout/library/std/src/sync/mpmc/utils.rs" "/checkout/library/std/src/sync/mpmc/error.rs" "/checkout/library/std/src/sync/mpmc/waker.rs" "/checkout/library/std/src/sync/mpmc/context.rs" "/checkout/library/std/src/sync/mpmc/array.rs" "/checkout/library/std/src/sync/mpmc/counter.rs" "/checkout/library/std/src/sync/mpmc/tests.rs" "/checkout/library/std/src/sync/mpmc/list.rs" "/checkout/library/std/src/sync/mpmc/mod.rs" "/checkout/library/std/src/sync/mpsc/sync_tests.rs" "/checkout/tests/run-make/native-lib-alt-naming/rust.rs" "/checkout/library/std/src/sync/mpsc/tests.rs" "/checkout/tests/run-make/native-lib-alt-naming/rmake.rs" "/checkout/library/std/src/sync/mpsc/mod.rs" "/checkout/tests/run-make/native-lib-alt-naming/native.rs" "/checkout/library/std/src/sync/once.rs" "/checkout/library/std/src/sync/lazy_lock.rs" "/checkout/library/std/src/sync/rwlock/tests.rs" "/checkout/library/std/src/sync/rwlock.rs" "/checkout/library/std/src/sync/once_lock.rs" "/checkout/library/std/src/sync/mod.rs" "/checkout/library/std/src/panic.rs" "/checkout/tests/run-make/macos-fat-archive/rmake.rs" "/checkout/tests/run-make/macos-fat-archive/lib.rs" "/checkout/library/std/src/f64/tests.rs" "/checkout/library/std/src/panic/tests.rs" "/checkout/library/std/src/num.rs" "/checkout/library/std/src/f64.rs" "/checkout/library/std/src/env/tests.rs" "/checkout/tests/run-make/emit-to-stdout/test.rs" "/checkout/tests/run-make/emit-to-stdout/rmake.rs" "/checkout/tests/run-make/rust-lld-by-default-nightly/main.rs" "/checkout/tests/run-make/rust-lld-by-default-nightly/rmake.rs" "/checkout/tests/run-make/optimization-remarks-dir/foo.rs" "/checkout/tests/run-make/optimization-remarks-dir/rmake.rs" "/checkout/library/std/src/collections/hash/set/tests.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-test/rmake.rs" "/checkout/library/std/src/collections/hash/map/tests.rs" "/checkout/library/std/src/collections/hash/set.rs" "/checkout/library/std/src/collections/hash/mod.rs" "/checkout/library/std/src/collections/hash/map.rs" "/checkout/library/std/src/collections/mod.rs" "/checkout/library/std/src/rt.rs" "/checkout/library/std/src/macros.rs" "/checkout/library/std/src/panicking.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-test/src/lib.rs" "/checkout/library/std/src/fs/tests.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-test/examples/ex.rs" "/checkout/library/std/src/hash/random.rs" "/checkout/library/std/src/hash/mod.rs" "/checkout/library/std/src/f128/tests.rs" "/checkout/tests/run-make/ice-static-mir/foo.rs" "/checkout/tests/run-make/ice-static-mir/bar.rs" "/checkout/tests/run-make/ice-static-mir/rmake.rs" "/checkout/tests/run-make/ice-static-mir/baz.rs" "/checkout/tests/run-make/rustc-macro-dep-files/foo.rs" "/checkout/tests/run-make/rustc-macro-dep-files/bar.rs" "/checkout/tests/run-make/rustc-macro-dep-files/rmake.rs" "/checkout/tests/run-make/split-debuginfo/main.rs" "/checkout/tests/run-make/split-debuginfo/foo.rs" "/checkout/tests/run-make/split-debuginfo/bar.rs" "/checkout/tests/run-make/include-bytes-deps/rmake.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Wed Nov  6 15:30:18 UTC 2024
  network time: Wed, 06 Nov 2024 15:30:18 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/std/src/f128/tests.rs
+++ b/library/std/src/f128/tests.rs
@@ -2,7 +2,10 @@
 #![cfg(reliable_f128)]
 
 use crate::f128::consts;
-use crate::num::{FpCategory as Fp, *};
+use crate::num::FpCategory as Fp;
+#[cfg(reliable_f128_math)]
+use crate::ops::Rem;
+use crate::ops::{Add, Div, Mul, Sub};
 
 // Note these tolerances make sense around zero, but not for more extreme exponents.
 
@@ -53,7 +56,22 @@ macro_rules! assert_f128_biteq {
 
 #[test]
 fn test_num_f128() {
-    test_num(10f128, 2f128);
+    // FIXME(f16_f128): replace with a `test_num` call once the required `fmodl`/`fmodf128`
+    // function is available on all platforms.
+    let ten = 10f128;
+    let two = 2f128;
+    assert_eq!(ten.add(two), ten + two);
+    assert_eq!(ten.sub(two), ten - two);
+    assert_eq!(ten.mul(two), ten * two);
+    assert_eq!(ten.div(two), ten / two);
+}
+
+#[test]
+#[cfg(reliable_f128_math)]
+fn test_num_f128_rem() {
+    let ten = 10f128;
+    let two = 2f128;
+    assert_eq!(ten.rem(two), ten % two);
 }
 
 #[test]

